pkg/translator/jaeger/jaegerproto_to_traces.go

```diff
@@ -302,7 +304,11 @@ func setInternalSpanStatus(attrs pcommon.Map, dest ptrace.Status) {
 		// Fallback to introspecting if this span represents a failed HTTP
 		// request or response, but again, only do so if the `error` tag was
 		// not set to true and no explicit status was sent.
-		if code, err := getStatusCodeFromHTTPStatusAttr(httpCodeAttr); err == nil {
+		// For HTTP status codes in the 4xx range span status MUST be left unset
+		// in case of SpanKind.SERVER and MUST be set to Error in case of SpanKind.CLIENT.
+		// For HTTP status codes in the 5xx range, as well as any other code the client
+		// failed to interpret, span status MUST be set to Error.
+		if code, err := getStatusCodeFromHTTPStatusAttr(httpCodeAttr); span.Kind() == ptrace.SpanKindClient && err == nil {
```

Comment:
I don't think it's correct logic here. Now we don't set error if span kind is server even for 5xx, but according to the spec we should, right? You can pass span kind to `getStatusCodeFromHTTPStatusAttr` and use it as condition only for 4xx codes.